src/fondant/pipeline/compiler.py

```diff
@@ -487,3 +487,133 @@ def _set_configuration(task, fondant_component_operation):
+            import sagemaker.workflow.pipeline
+            import sagemaker.workflow.steps
+
+            self.sagemaker = sagemaker
+
+        except ImportError:
+            msg = """You need to install the sagemaker extras to use the sagemaker compiler,\n
+                     you can install it with `pip install fondant[sagemaker]`"""
+            raise ImportError(
+                msg,
+            )
+
+    def compile(
+        self,
+        pipeline: Pipeline,
+        output_path: str,
+    ) -> None:
+        """Compile a fondant pipeline to sagemaker pipeline spec and save it
+        to a specified output path.
+
+        Args:
+            pipeline: the pipeline to compile
+            output_path: the path where to save the Kubeflow pipeline spec.
+        """
+        run_id = pipeline.get_run_id()
+        path = pipeline.base_path
+        pipeline.validate(run_id=run_id)
+
+        component_cache_key = None
+
+        steps: t.List[t.Any] = []
+        for component_name, component in pipeline._graph.items():
+            component_op = component["fondant_component_op"]
+
+            component_cache_key = component_op.get_component_cache_key(
+                previous_component_cache=component_cache_key,
+            )
+
+            metadata = Metadata(
+                pipeline_name=pipeline.name,
+                run_id=run_id,
+                base_path=path,
+                component_id=component_name,
+                cache_key=component_cache_key,
+            )
+
+            logger.info(f"Compiling service for {component_name}")
+
+            # add metadata argument to command
+            command = ["--metadata", metadata.to_json()]
+
+            # add in and out manifest paths to command
+            command.extend(
+                [
+                    "--output_manifest_path",
+                    f"{path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                    f"{component_name}/manifest.json",
+                ],
+            )
+
+            # add arguments if any to command
+            for key, value in component_op.arguments.items():
+                if isinstance(value, (dict, list)):
+                    command.extend([f"--{key}", json.dumps(value)])
+                else:
+                    command.extend([f"--{key}", f"{value}"])
+
+            # resolve dependencies
+            if component["dependencies"]:
+                for dependency in component["dependencies"]:
+                    # there is only an input manifest if the component has dependencies
+                    command.extend(
+                        [
+                            "--input_manifest_path",
+                            f"{path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                            f"{dependency}/manifest.json",
+                        ],
+                    )
+
+                    depends_on = [steps[-1]]
+            else:
+                depends_on = []
```

Comment:
This looks very similar to the other compilers. Do you think it makes sense to combine this?